Self-Destruct Rework

TerryTehCoder opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

Refactors the Self-Destruct system to acommodate more _unusual_ circumstances. Effects are split into different profile datums, profile datums register signals to a controller, the controller handles its own process loop with an internal timer. 

This allows us to do things like: 
-Specific Milestones which trigger at a given (absolute/relative) point in the countdown.
-Easier integration of varied self-destruct systems; not everything needs or has to be a nuclear bomb. 
-More easily find specific mechanical effects related to the self-destruct system.


https://github.com/user-attachments/assets/20f6ee73-edd5-489f-88d7-9228d0a0f7f5



## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

Supports modularity, interesting mechanical effects, and expansion. 

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
refactor: Reworked the Self-Destruct system to be more modular.
Added: New sound/lighting effects for default self-destruct process. 
Ports: Some sounds like Delta alert, poweron/poweroff from the old repository. 
fix: Changes the Self-Destruct cinematic to Title3, the SCP logo on a black background. (Stand-In replacement for Station blowing up)
balance: Self-Destruct minimum time is now 5 minutes.
\:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

TerryTehCoder committed:
Backup (Regular Spinny Lights Working)

TerryTehCoder committed:
Signals Refactor

TerryTehCoder committed:
Backup/Fixing up Milestones

TerryTehCoder committed:
BACKUP

Audio Milestones Working, Lights Working (Clearing on cancel too), controller ticking working [for the most part]. Sounds updated to loop, and be nicer.

TerryTehCoder committed:
Timer Fixes backup

TerryTehCoder committed:
Cleanup

Sounds/spin effects del when lights break/del

TerryTehCoder committed:
Cinematic/Final Sounds/Absolute Minimum

TerryTehCoder committed:
Timer default to 5 minutes.

TerryTehCoder commented:
This sounds silly but I need to revisit how the actual detonation is occurring. I don't think the current invoke for killing everyone in the Z is setup correctly in my refactor.

TerryTehCoder committed:
Fixes Explosion Logic

Missed a coordinate evaluation in the bomb location calculation which was preventing it from flagging as on_station and not allowing gibbing of people.

TerryTehCoder commented:
Detonation logic is fixed. Missed something when moving over to the explosion profile. Good for review now.

Tsurupeta commented on code/controllers/subsystem/processing/processing.dm:
```diff
@@ -32,7 +32,7 @@ SUBSYSTEM_DEF(processing)
 		if(QDELETED(thing))
 			processing -= thing
-		else if(call(thing, TYPE_PROC_REF(/datum, process))(wait * 0.1) == PROCESS_KILL)
+		else if(call(thing, TYPE_PROC_REF(/datum, process))(wait) == PROCESS_KILL)
```
Why? From the description of the process proc, it appears that they expect the number to be "seconds", not "deciseconds".

TerryTehCoder replied:
Looks like it might have been a mistake from when I was troubleshooting some of the processing logic early on. Good catch.

TerryTehCoder committed:
Updates processing.dm

Accidentally commited a change while troubleshooting some timing logic that was not intended.

TerryTehCoder committed:
Cleanup

Moves time decrement to after milestone check, so time 1 relative milestones actually fire. Makes Delta announcement milestone so we can actually position it After the first trill goes off.

Trims Delta audio mentioning countdown time because we won't Always use a 15 minute timer.

tichys reviewed (approved):
Looks good too me

TerryTehCoder committed:
Removes erroneous comment and cleans up conflict.

Reorders sound channels so we get rid of that nasty conflict.

TerryTehCoder committed:
Merge branch 'master' into Self-Destruct-Rework

bddvlpr commented:
<img width="661" height="304" alt="cooltext421899515429879" src="https://github.com/user-attachments/assets/2f6ed27b-c904-45fa-b63c-1b1aed93c11c" />
